FosServer/src/main/java/com/example/fostracker/servlets/FosTrackerServlet/GetNumberOfStoresByStatus.java

```diff
@@ -0,0 +1,100 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.example.fostracker.servlets.FosTrackerServlet;
+
+import com.example.fostracker.models.Coordinates;
+import com.example.fostracker.models.Agent;
+import com.google.gson.Gson;
+import com.example.fostracker.servlets.AgentServlet.SpannerQueryFunctions;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+
+import com.google.cloud.spanner.ResultSet;
+
+/**
+ * Prints the total number of stores by verification status in JSON format.
+ * <p>
+ * This Servlet converts the ResultSet into JSON object and prints to the response on @WebServlet(value = "/number_of_stores_by_status").
+ */
+
+@WebServlet(value = "/number_of_stores_by_status")
+public class GetNumberOfStoresByStatus extends HttpServlet {
+
+    // Gson object that is used to convert Strings into JSON objects
+    private Gson gson = new Gson();
+
+    /**
+     * HTTP Get method prints the query as response.
+     *
+     * @param request                         is GET request.
+     * @param response                        is HttpServletResponse object that is used to write the response.
+     * @throws ServletException
+     * @throws IOException
+     */
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        // sets the response to json format.
+        response.setContentType("application/json");
+        response.setCharacterEncoding("UTF-8");
+
+        // output variable is used to write to response.
+        PrintWriter output;
+        output = response.getWriter();
+
+        // Calling function to query the database and saving the result in numberOfMerchantsByStatus
+        ResultSet numberOfStoresByStatus = SpannerQueryFunctions.getNumberOfStoresByStatus();
+
+        // HashMap for storing the number of merchants in each status category.
+        HashMap<String, Long> statusToNumberOfStoresMap = new HashMap<>();
+
+        // If data is not empty then prints all the rows else prints "No data exists".
+        if (numberOfStoresByStatus.next()) {
+
+            // Loop through all rows and key as verification status and value as number of stores in hashmap.
+            do {
+                Long value = numberOfStoresByStatus.getLong("NumberOfStores");
+                String key = "UNVISITED";
+                if (numberOfStoresByStatus.isNull("VerificationStatus") == false){
+                    key = numberOfStoresByStatus.getString("VerificationStatus");
+                }
+                statusToNumberOfStoresMap.put(key, value);
+
+            } while (numberOfStoresByStatus.next());
+
+
```

Comment:
Does formatter remove these whitespaces?

Reply:
It did not remove these spaces. But now I've removed them manually